stjude/proteinpaint
# Gdc await catch

## Description

This update attempts to expand auto-recovery for recoverable errors besides the initial healthy GDC API detection.  Previously, retries only applied to detecting a healthy `/status` response, and the server crashed if any subsequent caching requests failed due to connectivity issues or API 5xx status (server maintenance). 

To test:
- set `serverconfig.features.gdcCacheCheckWait=9000` 
- use `npx tsx server.ts` from sjpp instead of `npm start` to more clearly see server crash
- and uncomment one of the test `.then()` lines in `initGdc.cache.js`
  - line ~136: HTTP 4xx should cause an immediate crash on initial cache attempt, client request must be fixed
  - line ~450: HTTP 5xx should not cause a crash, API may be under maintenance and will be healthy again
  - line ~552: error code='ENOTFOUND' is assumed to be caused by a temporary network issue that will be resolved
  - line ~82: one of the preceding `.then()` recoverable errors has to be active, which will be converted to a non-recoverable error on retry: should cancel all subsequent retries

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
